pkg/virtctl/credentials/add-key/add-key.go

```diff
@@ -0,0 +1,280 @@
+		if deleteErr != nil && err == nil {
+			// Return the deletion error from the enclosing function
+			err = fmt.Errorf("error deleting secret: %w", deleteErr)
+		}
+	}()
+
+	accessCredential := newAccessCredential(secret.Name, cmdFlags.User)
+	accessCredentialPatch := patchToAddAccessCredential(accessCredential)
+
+	// First, try patch that adds the array
+	fullPatch := common.MustMarshalPatch(append(patchToAddAccessCredentialsArray(), accessCredentialPatch)...)
+	_, err = cli.VirtualMachine(vm.Namespace).Patch(cmd.Context(), vm.Name, types.JSONPatchType, fullPatch, &metav1.PatchOptions{})
+	if err != nil {
+		// If it fails, it probably means that the array exists.
+		// Try to add the new access credential to the existing array.
+		_, err = cli.VirtualMachine(vm.Namespace).Patch(cmd.Context(), vm.Name, types.JSONPatchType, common.MustMarshalPatch(accessCredentialPatch), &metav1.PatchOptions{})
+		if err != nil {
+			return fmt.Errorf("error patching virtual machine: %w", err)
+		}
+	}
+
+	vmPatched = true
+	return nil
+}
+
+func updateSecretWithSshKey(cmd *cobra.Command, cli kubecli.KubevirtClient, cmdFlags *addSshKeyFlags, vm *v1.VirtualMachine, sshKey string) error {
+	secrets := common.GetSshSecretsForUser(vm.Spec.Template.Spec.AccessCredentials, cmdFlags.User)
+	if len(secrets) == 0 {
+		return fmt.Errorf("no secrets specified for user: %s", cmdFlags.User)
+	}
+
+	secretName, err := common.FindSecretOrGetFirst(cmdFlags.Secret, secrets)
+	if err != nil {
+		return err
+	}
+
+	secret, err := cli.CoreV1().Secrets(vm.Namespace).Get(cmd.Context(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error getting secret \"%s\": %w", secretName, err)
+	}
+
+	if secretContainsKey(secret.Data, sshKey) {
+		cmd.Printf("Secret \"%s\" already contains this SSH key.", secretName)
+		return nil
+	}
+
+	if !cmdFlags.Force {
+		// Check if secret is owned by the VM. This is useful to not accidentally update a secret that is used by multiple VMs.
+		var isOwnedByVm bool
+		for _, ownerReference := range secret.OwnerReferences {
+			if ownerReference.Kind == v1.VirtualMachineGroupVersionKind.Kind &&
+				ownerReference.Name == vm.Name &&
+				ownerReference.UID == vm.UID {
+
+				isOwnedByVm = true
+				break
+			}
+		}
+		if !isOwnedByVm {
+			return fmt.Errorf("secret %s does not have an owner reference pointing to VM %s", secretName, vm.Name)
+		}
+	}
+
+	randomFilename := "ssh-key-" + rand.String(6)
+	addKeyPatch := common.AddKeyToSecretPatchOp(randomFilename, []byte(sshKey))
+
+	// First try patch with adding the /data field
+	fullPatch := common.MustMarshalPatch(append(common.AddDataFieldToSecretPatchOp(), addKeyPatch)...)
+	_, err = cli.CoreV1().Secrets(vm.Namespace).Patch(cmd.Context(), secretName, types.JSONPatchType, fullPatch, metav1.PatchOptions{})
+	if err != nil {
+		// If full patch fails, it probably means that /data field exists.
```

Comment:
This will likely be the case most times, so switch the order of patches?

Reply:
done